src/lib.rs

```diff
@@ -41,6 +45,37 @@ fn validate(payload: &[u8]) -> CallResult {
     let validation_request: ValidationRequest<Settings> = ValidationRequest::new(payload)?;
     let deployment = serde_json::from_value::<Deployment>(validation_request.request.object)?;
 
+    let pod = deployment.clone().spec.unwrap().template.spec.unwrap();
+    let service_account = pod
+        .clone()
+        .service_account_name
+        .or(pod.service_account)
+        .unwrap_or_default();
+    let sar = SubjectAccessReview {
+        spec: SubjectAccessReviewSpec {
+            user: Some(service_account.to_owned()),
+            resource_attributes: Some(ResourceAttributes {
+                namespace: Some("default".to_string()),
```

Comment:
```suggestion
                namespace: Some("kube-system".to_string()),
```

Maybe it's better to check for not having permissions to create Pods on `kube-system` or something similar, instead of `default`, feels more usual (sorry for the nitpicking).